integration/spark2/src/main/scala/org/apache/spark/sql/events/MergeIndexEventListener.scala

```diff
@@ -23,61 +23,22 @@ import scala.collection.JavaConverters._
 import scala.collection.mutable
 
 import org.apache.spark.internal.Logging
-import org.apache.spark.SparkContext
 import org.apache.spark.rdd.CarbonMergeFilesRDD
-import org.apache.spark.sql.SparkSession
 import org.apache.spark.sql.util.CarbonException
+import org.apache.spark.util.MergeIndexUtil
 
 import org.apache.carbondata.common.logging.LogServiceFactory
-import org.apache.carbondata.core.constants.CarbonCommonConstants
 import org.apache.carbondata.core.datamap.Segment
 import org.apache.carbondata.core.locks.{CarbonLockFactory, LockUsage}
-import org.apache.carbondata.core.metadata.SegmentFileStore
-import org.apache.carbondata.core.metadata.schema.table.CarbonTable
 import org.apache.carbondata.core.statusmanager.SegmentStatusManager
-import org.apache.carbondata.events.{AlterTableCompactionPostEvent, AlterTableMergeIndexEvent, Event, OperationContext, OperationEventListener}
-import org.apache.carbondata.processing.loading.events.LoadEvents.LoadTablePostExecutionEvent
+import org.apache.carbondata.events._
 import org.apache.carbondata.processing.merger.CarbonDataMergerUtil
 
 class MergeIndexEventListener extends OperationEventListener with Logging {
   val LOGGER = LogServiceFactory.getLogService(this.getClass.getCanonicalName)
 
   override def onEvent(event: Event, operationContext: OperationContext): Unit = {
     event match {
-      case preStatusUpdateEvent: LoadTablePostExecutionEvent =>
```

Comment:
Basically, we need to merge the merge index files before commiting, so can you add preTableStatusLoadEvent and load the merge index there?

Reply:
yes.. for LOAD we can merge index in the preStatusUpdate Event. Done.